sympy/physics/mechanics/kane.py

```diff
@@ -303,6 +323,26 @@ def _initialize_kindiffeq_matrices(self, kdeqs):
             self._k_ku = Matrix()
             self._k_kqdot = Matrix()
 
+    def _partial_velocity(self, vel_list):
+        '''
+        Return the partial velocity w.r.t. generalized speeds
+        Throws an error if qdot terms are left
+        '''
+        self._assert_no_qdot(vel_list)
+        return partial_velocity(vel_list, self.u, self._inertial)
+
+    def _assert_no_qdot(self, vel_or_acc_list):
+        q_dot_set = set(self._qdot).union([_qd.diff() for _qd in self._qdot])
+        for vel in vel_or_acc_list:
+            dyn_symbols = find_dynamicsymbols(vel, reference_frame=self._inertial)
```

Comment:
This could be a very slow operation for very large systems. I'm not a fan of adding this. There must be a faster way to check if qdots are present. I bet differentiating wrt to the qdots is faster than this.

Reply:
gotcha - I assume it's the `find_dynamicssymbols` that you expect to be slow? I'll see what I can do with using diff() instead

Reply:
I tried this with using diff and jacobian, and found actually that it was slower than using find_dynamicssymbols (10ms vs 1ms tested on the disc linearization test).
I left it in the code for now just to show what it looks like. Let me know if you have suggestions there for something faster, otherwise I think find_dynamicssymbols is sufficiently fast?

Reply:
It currently fires for both cases.


I don't have a good handle for the complexity of `find_dynamicssymbols`: in the project I've used it with 30+ states I haven't encountered issues. My approach would be to keep the logic as is and only address it if someone raises an issue, but if your intuition says this will be slow for uses cases you know people might be interested in, then it's easy enough to shortcut it by returning immediately if `self.explicit_kinematics == True`.

I'm fine with either options, let me know what you prefer and I'll push one or the other.

Reply:
The place were I found find_dynamicsymbols to be slow was using it on this problem https://pydy.readthedocs.io/en/latest/examples/carvallo-whipple.html. For example it takes 1.6 seconds to run it on fr + frstar on my computer. Running it on the n_link_pendulum_on_cart model in the mechanics package here with n=2 is like a 1/10 to 1/20th of a second, so that isn't bad. I'm just wary of introducing find_dynamicsymbols into the core code because we typically know what symbols we need a priori.